cryptosuite/bbsplussignaturesuite.go

```diff
@@ -0,0 +1,355 @@
+	}
+
+	// 4 & 5. create the signature over the provable data as a JWS
+	proofValue, err := s.Sign(tbs)
+	if err != nil {
+		return errors.Wrap(err, "could not sign provable value")
+	}
+
+	// set the signature on the proof object and return
+	proof.SetProofValue(base64.RawStdEncoding.EncodeToString(proofValue))
+	genericProof := crypto.Proof(proof)
+	p.SetProof(&genericProof)
+	return nil
+}
+
+func (b BBSPlusSignatureSuite) prepareProof(proof crypto.Proof, opts *ProofOptions) (*crypto.Proof, error) {
+	proofBytes, err := json.Marshal(proof)
+	if err != nil {
+		return nil, err
+	}
+
+	var genericProof map[string]interface{}
+	if err = json.Unmarshal(proofBytes, &genericProof); err != nil {
+		return nil, err
+	}
+
+	// proof cannot have a proof value
+	delete(genericProof, "proofValue")
+
+	// make sure the proof has a timestamp
+	created, ok := genericProof["created"]
+	if !ok || created == "" {
+		genericProof["created"] = GetRFC3339Timestamp()
+	}
+
+	var contexts []interface{}
+	if opts != nil {
+		contexts = opts.Contexts
+	} else {
+		// if none provided, make sure the proof has a context value for this suite
+		contexts = ArrayStrToInterface(b.RequiredContexts())
+	}
+	genericProof["@context"] = contexts
+	p := crypto.Proof(genericProof)
+	return &p, nil
+}
+
+func (b BBSPlusSignatureSuite) Verify(v Verifier, p Provable) error {
+	proof := p.GetProof()
+	gotProof, err := BBSPlusProofFromGenericProof(*proof)
+	if err != nil {
+		return errors.Wrap(err, "could not prepare proof for verification; error coercing proof into BBSPlusSignature2020Proof proof")
+	}
+
+	// remove proof before verifying
+	p.SetProof(nil)
+
+	// make sure we set it back after we're done verifying
+	defer p.SetProof(proof)
+
+	// remove the proof value in the proof before verification
+	signatureValue, err := decodeProofValue(gotProof.ProofValue)
+	if err != nil {
+		return errors.Wrap(err, "could not decode proof value")
+	}
+	gotProof.SetProofValue("")
+
+	// prepare proof options
+	contexts, err := GetContextsFromProvable(p)
+	if err != nil {
+		return errors.Wrap(err, "could not get contexts from provable")
+	}
+
+	// make sure the suite's context(s) are included
+	contexts = ensureRequiredContexts(contexts, b.RequiredContexts())
+	opts := &ProofOptions{Contexts: contexts}
+
+	// run CVH on both provable and the proof
+	tbv, err := b.CreateVerifyHash(p, gotProof, opts)
+	if err != nil {
+		return errors.Wrap(err, "create verify hash algorithm failed")
+	}
+
+	if err = v.Verify(tbv, signatureValue); err != nil {
+		return errors.Wrap(err, "could not verify BBS+ signature")
+	}
+	return nil
+}
+
+// decodeProofValue because the proof could have been encoded in a variety of manners we must try them all
+func decodeProofValue(proofValue string) ([]byte, error) {
+	signatureBytes, err := base64.RawStdEncoding.DecodeString(proofValue)
+	if err == nil {
+		return signatureBytes, nil
+	}
+	signatureBytes, err = base64.StdEncoding.DecodeString(proofValue)
```

Comment:
I've seen RawStdEncoding used everywhere else. Is this what you meant? 
```suggestion
	signatureBytes, err = base64.RawStdEncoding.DecodeString(proofValue)
```

Reply:
unfortunately, no. the aries impl uses StdEncoding. There's an issue to decide what to do. the decode functions handle both cases. this is required to get test vectors to pass. I'll link the issue https://github.com/w3c-ccg/ldp-bbs2020/issues/16#issuecomment-1436148820